financial-connections/src/main/java/com/stripe/android/financialconnections/features/networkinglinkverification/NetworkingLinkVerificationViewModel.kt

```diff
@@ -0,0 +1,149 @@
+import com.stripe.android.financialconnections.domain.PollNetworkedAccounts
+import com.stripe.android.financialconnections.domain.StartVerification
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.Pane
+import com.stripe.android.financialconnections.model.PartnerAccountsList
+import com.stripe.android.financialconnections.ui.FinancialConnectionsSheetNativeActivity
+import com.stripe.android.uicore.elements.IdentifierSpec
+import com.stripe.android.uicore.elements.OTPController
+import com.stripe.android.uicore.elements.OTPElement
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+internal class NetworkingLinkVerificationViewModel @Inject constructor(
+    initialState: NetworkingLinkVerificationState,
+    private val eventTracker: FinancialConnectionsAnalyticsTracker,
+    private val getManifest: GetManifest,
+    private val lookupAccount: LookupAccount,
+    private val startVerification: StartVerification,
+    private val confirmVerification: ConfirmVerification,
+    private val markLinkVerified: MarkLinkVerified,
+    private val pollNetworkedAccounts: PollNetworkedAccounts,
+    private val goNext: GoNext,
+    private val logger: Logger
+) : MavericksViewModel<NetworkingLinkVerificationState>(initialState) {
+
+    init {
+        logErrors()
+        suspend {
+            val email = requireNotNull(getManifest().accountholderCustomerEmailAddress)
+            val consumerSession = requireNotNull(lookupAccount(email).consumerSession)
+            startVerification(consumerSession.clientSecret)
+            eventTracker.track(PaneLoaded(Pane.NETWORKING_LINK_VERIFICATION))
+            NetworkingLinkVerificationState.Payload(
+                email = consumerSession.emailAddress,
+                phoneNumber = consumerSession.redactedPhoneNumber,
+                consumerSessionClientSecret = consumerSession.clientSecret,
+                otpElement = OTPElement(
+                    IdentifierSpec.Generic("otp"),
+                    OTPController()
+                )
+            )
+        }.execute { copy(payload = it) }
+    }
+
+    private fun logErrors() {
+        onAsync(
+            NetworkingLinkVerificationState::payload,
+            onSuccess = {
+                viewModelScope.launch {
+                    it.otpElement.getFormFieldValueFlow()
+                        .mapNotNull { formFieldsList ->
+                            // formFieldsList contains only one element, for the OTP. Take the second value of
+                            // the pair, which is the FormFieldEntry containing the value entered by the user.
+                            formFieldsList.firstOrNull()?.second?.takeIf { it.isComplete }?.value
+                        }.collectLatest { onOTPEntered(it) }
```

Comment:
took this from Link's OTP - https://github.com/stripe/stripe-android/blob/3c976825/link/src/main/java/com/stripe/android/link/ui/verification/VerificationViewModel.kt#L63